core/src/main/scala/org/apache/spark/status/AppStatusListener.scala

```diff
@@ -645,8 +645,15 @@ private[spark] class AppStatusListener(
   }
 
   override def onTaskEnd(event: SparkListenerTaskEnd): Unit = {
-    // TODO: can this really happen?
-    if (event.taskInfo == null) {
+    // TODO: can taskInfo null really happen?
```

Comment:
To answer this question, no `taskInfo` cannot be `null`  IMO - can you also take a look at it @Ngone51 ?
We can remove this check and the TODO once we are sure.

Reply:
Yea. I checked. Null taskInfo is not possible.